db/migrations/72_add_approval_related_columns.sql

```diff
@@ -0,0 +1,56 @@
+-- +migrate Up
+ALTER TABLE `groups`
+    ADD COLUMN `require_personal_info_access_approval` ENUM('none', 'view', 'edit') NOT NULL DEFAULT 'none'
+        AFTER `lock_user_deletion_until`,
+    ADD COLUMN `require_lock_membership_approval_until` DATETIME DEFAULT NULL
+        AFTER `require_personal_info_access_approval`,
+    ADD COLUMN `require_watch_approval` TINYINT(1) NOT NULL DEFAULT 0
+        AFTER `require_lock_membership_approval_until`;
+
+ALTER TABLE `groups_groups`
+    ADD COLUMN `personal_info_access_approved_at` DATETIME DEFAULT NULL AFTER `expires_at`,
+    ADD COLUMN `personal_info_access_approved` TINYINT(1)
+        AS (IFNULL(`personal_info_access_approved_at`, 0)) NOT NULL
```

Comment:
Are you sure `(IFNULL(personal_info_access_approved_at, 0))` is a better to compute `(personal_info_access_approved_at IS NOT NULL)`?
For me this looks like writing `if x == true return true; else return false` instead of `return x` directly.
Here it even has to cast the date to a int if not null... so it would return false even if it is not null but a date which equals "0", which I would expect to return TRUE. Probably it cannot happen anyway ;-) but just to say that it looks less readable and even "a bit less correct" (whatever it means;-)).

Ah there is even a `ISNULL()` fct in MySQL, didn't you want to that one? `NOT ISNULL(personal_info_access_approved_at)`

Maybe I just missed the reason why you did that.

Reply:
fixed